Explorer/Assets/DCL/SDKComponents/Tween/Systems/TweenUpdaterSystem.cs

```diff
@@ -113,138 +117,99 @@ private void HandleComponentRemoval(ref SDKTweenComponent tweenComponent, ref CR
-                        PrimitivesConversionExtensions.PBVectorToUnityVector(tweenModel.Scale.End),
-                        durationInSeconds, ease);
-
-                    break;
-                case PBTween.ModeOneofCase.Move:
-                default:
-                    tempTweener = SetupPositionTween(transformComponent.Transform,
-                        PrimitivesConversionExtensions.PBVectorToUnityVector(tweenModel.Move.Start),
-                        PrimitivesConversionExtensions.PBVectorToUnityVector(tweenModel.Move.End),
-                        durationInSeconds, ease, tweenModel.Move.HasFaceDirection && tweenModel.Move.FaceDirection);
-
-                    break;
+                UpdateTweenPosition(sdkEntity, sdkTweenComponent, ref sdkTransform);
             }
-
-            tempTweener.Goto(tweenModel.CurrentTime * durationInSeconds, isPlaying);
         }
 
-        private void CleanUpTweenBeforeRemoval(CRDTEntity sdkEntity, SDKTweenComponent sdkTweenComponent)
+        private void UpdateTweenStateAndPosition(CRDTEntity sdkEntity, SDKTweenComponent sdkTweenComponent, ref SDKTransform sdkTransform)
         {
-            sdkTweenComponent.Tweener.Kill();
-            ecsToCRDTWriter.DeleteMessage<PBTweenState>(sdkEntity);
+            TweenSDKComponentHelper.WriteTweenStateInCRDT(ecsToCRDTWriter, sdkEntity, sdkTweenComponent.TweenStateStatus);
+            UpdateTweenPosition(sdkEntity, sdkTweenComponent, ref sdkTransform);
         }
 
-        private Tweener SetupPositionTween(Transform entityTransform, Vector3 startPosition,
-            Vector3 endPosition, float durationInSeconds, Ease ease, bool faceDirection)
+        private void UpdateTweenPosition(CRDTEntity sdkEntity, SDKTweenComponent sdkTweenComponent, ref SDKTransform sdkTransform)
         {
-            if (faceDirection) entityTransform.forward = (endPosition - startPosition).normalized;
-
-            entityTransform.localPosition = startPosition;
-            return entityTransform.DOLocalMove(endPosition, durationInSeconds).SetEase(ease).SetAutoKill(false);
+            TweenSDKComponentHelper.WriteTweenResult(ref sdkTransform, sdkTweenComponent.CustomTweener);
+            TweenSDKComponentHelper.WriteTweenResultInCRDT(ecsToCRDTWriter, sdkEntity, sdkTweenComponent.CustomTweener);
         }
 
-        private TweenStateStatus GetCurrentTweenState(float currentTime, bool isPlaying)
+
+        private void SetupTweener(CRDTEntity entity, ref SDKTweenComponent sdkTweenComponent, Transform entityTransform, PBTween tweenModel, float durationInSeconds, bool isPlaying, SDKTransform sdkTransform)
         {
-            if (!isPlaying) { return TweenStateStatus.TsPaused; }
+            //NOTE: Left this per legacy reasons, Im not sure if this can happen in new renderer
+            // There may be a tween running for the entity transform, e.g: during preview mode hot-reload.
+            if (sdkTweenComponent.IsActive())
```

Comment:
According to the implementation of `IsActive` `CustomTweener != null && CustomTweener.IsActive();`:

if `CustomTweener` is assigned but is not active it will leak as you won't return it to the pool, and on line #L196 you override the value with a new object from the pool

Reply:
Great catch. Thanks, completely missed it